recipe/meta.yaml

```diff
@@ -34,35 +31,42 @@ requirements:
     - {{ compiler('cxx') }}
     # llvm is needed for the headers
     - llvm-openmp              # [osx]
-    - patch     # [not win]
-    - m2-patch  # [win]
+
   host:
     - python
     - pip
     - setuptools
     - wheel
-    - llvmlite 0.38.*
+    - llvmlite 0.39.*
     - numpy
-    - tbb-devel 2021.*
-    - _openmp_mutex            # [linux and not ppc64le]
+    - tbb-devel 2021.*,<2021.6
+    - _openmp_mutex            # [linux]
+    - importlib_metadata       # [py<39]
+
   run:
     - python
-    - {{ pin_compatible('llvmlite', max_pin='x.x') }}
-    - {{ pin_compatible('numpy') }}
+    - importlib_metadata       # [py<39]
+    - llvmlite >=0.39.*,<1
+    # NumPy has a hard upper limit.
+    # NumPy 1.22.0, 1.22.1, 1.22.2 are all broken for ufuncs
+    # xref: https://github.com/numba/numba/issues/7756
+    # NumPy >=1.23 breaks with CUDA test: test_reinterpret_array_type
+    # xref: https://github.com/numba/numba/issues/8529
+    - numpy >=1.18,!=1.22.0,!=1.22.1,!=1.22.2,<1.23
     # needed for pkg_resources
     - setuptools
 
   run_constrained:
     - tbb  >=2021.0
     # avoid confusion from openblas bugs
     - libopenblas !=0.3.6      # [x86_64]
-    # CUDA 9.2 or later is required for CUDA support
-    - cudatoolkit >=9.2   # [x86 and not osx]
-    - cudatoolkit >=9.0   # [x86 and osx]
+    - libopenblas >=0.3.18, !=0.3.20  # [arm64]
+    # CUDA 10.2 or later is required for CUDA support
+    - cudatoolkit >=10.2
     # scipy 1.0 or later
     - scipy >=1.0
-    # numpy has a hard upper limit currently
-    - numpy >=1.18,<1.22
+    # CUDA Python 11.6 or later
+    - cuda-python >=11.6
 
 test:
   requires:
```

Comment:
ipykernel is in numba's recipe:
https://github.com/numba/numba/blob/0.56.3/buildscripts/condarecipe.local/meta.yaml#L85

pygments and pexpect could also be added:
https://github.com/numba/numba/blob/0.56.3/buildscripts/incremental/setup_conda_environment.sh#L62

tbb also has an upper bound:
https://github.com/numba/numba/blob/0.56.3/buildscripts/incremental/setup_conda_environment.sh#L107

Reply:
Sure, `ipykernel`, `pygments`, and `pexpect` can be added, I can see they're used in some optional tests.

My `tbb` comments are here: https://github.com/AnacondaRecipes/numba-feedstock/pull/10#discussion_r1004014248